### PR TITLE
Fix some more warnings

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

**Comment by Bond-009** [comment]
@dkanada Yes

File: MediaBrowser.Providers/Subtitles/SubtitleManager.cs [hunk 1]
```diff
@@ -48,17 +47,18 @@ public class SubtitleManager : ISubtitleManager
...
+        /// <inheritdoc />
         public void AddParts(IEnumerable<ISubtitleProvider> subtitleProviders)
         {
             _subtitleProviders = subtitleProviders
-                .OrderBy(i =>
-                {
-                    var hasOrder = i as IHasOrder;
-                    return hasOrder == null ? 0 : hasOrder.Order;
-                })
+                .OrderBy(i => !(i is IHasOrder hasOrder) ? 0 : hasOrder.Order)
```
Comment (JustAMan) [inline MediaBrowser.Providers/Subtitles/SubtitleManager.cs]:
```suggestion
                .OrderBy(i => (i is IHasOrder hasOrder) ? hasOrder.Order : 0)
```
IMHO it's easier to understand this way

File: MediaBrowser.Providers/Subtitles/SubtitleManager.cs [hunk 2]
```diff
@@ -19,6 +19,7 @@
 using MediaBrowser.Model.IO;
 using MediaBrowser.Model.Providers;
 using Microsoft.Extensions.Logging;
+using static MediaBrowser.Model.IO.StreamDefaults;
```
Comment (JustAMan) [inline MediaBrowser.Providers/Subtitles/SubtitleManager.cs]: why is this suddenly needed?

File: MediaBrowser.Providers/TV/DummySeasonProvider.cs [hunk 3]
```diff
@@ -127,17 +127,24 @@ private async Task<bool> AddDummySeasonFolders(Series series, CancellationToken
...
-                _libraryManager.GetLibraryOptions(series).SeasonZeroDisplayName :
-                (seasonNumber.HasValue ? string.Format(_localization.GetLocalizedString("NameSeasonNumber"), seasonNumber.Value.ToString(_usCulture)) : _localization.GetLocalizedString("NameSeasonUnknown"));
+            var seasonName = seasonNumber == 0
+                ? _libraryManager.GetLibraryOptions(series).SeasonZeroDisplayName
+                : (seasonNumber.HasValue
+                    ? string.Format(
+                        CultureInfo.InvariantCulture,
+                        _localization.GetLocalizedString("NameSeasonNumber"),
+                        seasonNumber.Value)
+                    : _localization.GetLocalizedString("NameSeasonUnknown"));
```
Comment (JustAMan) [inline MediaBrowser.Providers/TV/DummySeasonProvider.cs]: this looks like uglified JS we're battling with... :) can this huge ternary hell be replaced with nice chain of `if`-s?

File: Emby.Server.Implementations/Data/SqliteExtensions.cs [hunk 4]
```diff
@@ -9,6 +9,47 @@ namespace Emby.Server.Implementations.Data
...
+            "yyyyMMddHHmmss",
+            "yyyyMMddHHmm",
+            "yyyyMMddTHHmmssFFFFFFF",
+            "yyyy-MM-dd",
+            "yyyyMMdd",
+            "yy-MM-dd"
+        };
+
+        private static readonly string _datetimeFormatUtc = _datetimeFormats[5];
+        private static readonly string _datetimeFormatLocal = _datetimeFormats[19];
```
[the remark this answers is not in the document]
Comment (Bond-009) [reply Emby.Server.Implementations/Data/SqliteExtensions.cs]: Yeah, that does seem way more logical, but this PR just moved these fields

[the remark this answers is not in the document]
Comment (JustAMan) [reply Emby.Server.Implementations/Data/SqliteExtensions.cs]: So what? Why cannot this fix be made in?

File: MediaBrowser.Providers/Subtitles/SubtitleManager.cs (hunk 2 above)
Comment (Bond-009) [reply MediaBrowser.Providers/Subtitles/SubtitleManager.cs]: For the ‘DefaultFileStreamBufferSize’ constant

File: Emby.Server.Implementations/LiveTv/LiveTvManager.cs [hunk 5]
```diff
@@ -1226,12 +1226,13 @@ private async Task RefreshChannelsInternal(IProgress<double> progress, Cancellat
                         currentChannel.AddTag("Kids");
                     }
 
-                    //currentChannel.UpdateToRepository(ItemUpdateType.MetadataImport, cancellationToken);
-                    await currentChannel.RefreshMetadata(new MetadataRefreshOptions(new DirectoryService(_logger, _fileSystem))
-                    {
-                        ForceSave = true
-
-                    }, cancellationToken).ConfigureAwait(false);
+                    currentChannel.UpdateToRepository(ItemUpdateType.MetadataImport, cancellationToken);
```
[the remark this answers is not in the document]
Comment (Bond-009) [reply Emby.Server.Implementations/LiveTv/LiveTvManager.cs]: It was commented out by the person why ported Emby to .Net Core

File: MediaBrowser.Common/Cryptography/PasswordHash.cs [hunk 6]
```diff
@@ -122,10 +122,10 @@ private void SerializeParameters(StringBuilder stringBuilder)
             stringBuilder.Append('$');
             foreach (var pair in _parameters)
             {
-                stringBuilder.Append(pair.Key);
-                stringBuilder.Append('=');
-                stringBuilder.Append(pair.Value);
-                stringBuilder.Append(',');
+                stringBuilder.Append(pair.Key)
+                    .Append('=')
```
[the remark this answers is not in the document]
Comment (Bond-009) [reply MediaBrowser.Common/Cryptography/PasswordHash.cs]: Sometimes I forget what I've already changed in another PR. Not very efficient :/

File: Emby.Server.Implementations/LiveTv/LiveTvManager.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (JustAMan) [reply Emby.Server.Implementations/LiveTv/LiveTvManager.cs]: That still doesn't explain why it was commented... but let's try and see what breaks if we enable it :)

**Comment by JustAMan** [comment]
@Bond-009 you now have a conflict in this PR

**Comment by Bond-009** [comment]
@JustAMan fixed

File: MediaBrowser.Providers/Music/AudioDbArtistProvider.cs [hunk 7]
```diff
@@ -141,22 +145,22 @@ internal async Task DownloadArtistInfo(string musicBrainzId, CancellationToken c
...
-                CancellationToken = cancellationToken,
-                BufferContent = true
+            using (var httpResponse = await _httpClient.SendAsync(
+                new HttpRequestOptions
+                {
+                    Url = url,
+                    CancellationToken = cancellationToken,
+                    BufferContent = true
 
-            }, "GET").ConfigureAwait(false))
```
Comment (dkanada) [inline MediaBrowser.Providers/Music/AudioDbArtistProvider.cs]: we don't need the newline on 154 here

File: MediaBrowser.Providers/Music/AudioDbExternalIds.cs [hunk 8]
```diff
@@ -6,58 +6,63 @@ namespace MediaBrowser.Providers.Music
...
 
+        /// <inheritdoc />
         public string UrlFormatString => "https://www.theaudiodb.com/artist/{0}";
 
+        /// <inheritdoc />
         public bool Supports(IHasProviderIds item)
-        {
-            return item is Audio || item is MusicAlbum;
-        }
+         => item is Audio || item is MusicAlbum;
```
Comment (dkanada) [inline MediaBrowser.Providers/Music/AudioDbExternalIds.cs]: indentation is off

File: MediaBrowser.Providers/Music/MusicBrainzAlbumProvider.cs [hunk 9]
```diff
@@ -14,24 +15,13 @@
...
     {
-        internal static MusicBrainzAlbumProvider Current;
-
-        private readonly IHttpClient _httpClient;
-        private readonly IApplicationHost _appHost;
-        private readonly ILogger _logger;
-        private readonly IJsonSerializer _json;
-        private Stopwatch _stopWatchMusicBrainz = new Stopwatch();
-
-        public readonly string MusicBrainzBaseUrl;
```
Comment (dkanada) [inline MediaBrowser.Providers/Music/MusicBrainzAlbumProvider.cs]: Something in this file is breaking the syntax highlighting, any clue why?

File: MediaBrowser.Providers/Music/MusicExternalIds.cs [hunk 10]
```diff
@@ -7,99 +7,113 @@ namespace MediaBrowser.Providers.Music
...
     {
+        /// <inheritdoc />
         public string Name => "MusicBrainz Artist";
 
+        /// <inheritdoc />
+
         public string Key => MetadataProviders.MusicBrainzArtist.ToString();
 
+        /// <inheritdoc />
+
```
Comment (dkanada) [inline MediaBrowser.Providers/Music/MusicExternalIds.cs]: extra newline

File: MediaBrowser.Providers/Music/MusicBrainzAlbumProvider.cs (hunk 9 above)
Comment (Bond-009) [reply MediaBrowser.Providers/Music/MusicBrainzAlbumProvider.cs]: No, looks fine

**Comment by JustAMan** [comment]
@Bond-009 why is CI failing?

File: MediaBrowser.Providers/TV/DummySeasonProvider.cs [hunk 11]
```diff
@@ -130,7 +130,7 @@ private async Task<bool> AddDummySeasonFolders(Series series, CancellationToken
             string seasonName;
             if (seasonNumber == null)
             {
-                seasonName =_localization.GetLocalizedString("NameSeasonUnknown")
+                seasonName =_localization.GetLocalizedString("NameSeasonUnknown");
```
Comment (JustAMan) [inline MediaBrowser.Providers/TV/DummySeasonProvider.cs]: shouldn't it be
```c#
seasonName = _localization.GetLocalizedString("NameSeasonUnknown");
```
?

Comment (Bond-009) [reply MediaBrowser.Providers/TV/DummySeasonProvider.cs]: Good catch

**Pull request merged**